src/lib/transition_frontier_controller/transition_frontier_controller.ml

```diff
@@ -40,6 +41,32 @@ let run ~logger ~trust_system ~verifier ~network ~time_controller
       Transition_handler.Unprocessed_transition_cache.register_exn
         unprocessed_transition_cache t
       |> Strict_pipe.Writer.write primary_transition_writer ) ;
+  let initial_state_hashes =
+    List.map collected_transitions ~f:(fun envelope ->
+        Network_peer.Envelope.Incoming.data envelope
+        |> Mina_transition.External_transition.Initial_validated.state_hash )
+  in
+  let extensions = Transition_frontier.extensions frontier in
+  let already_reported = ref false in
+  don't_wait_for
+  @@ Pipe_lib.Broadcast_pipe.Reader.iter
+       (Transition_frontier.Extensions.get_view_pipe extensions New_breadcrumbs)
+       ~f:(fun new_breadcrumbs ->
+         let new_state_hashes =
+           List.map new_breadcrumbs
+             ~f:Transition_frontier.Breadcrumb.state_hash
+         in
+         List.iter new_state_hashes ~f:(fun new_state_hash ->
+             if
+               (not !already_reported)
+               && List.mem initial_state_hashes new_state_hash
```

Comment:
I would agree that this metrics does not reflect the initial catchup time accurately.

My definition of initial catchup would be the time for a node to sync with the network after bootstrap.
The blocks collected during bootstrap would trigger initial catchup, but the target of initial catchup would move forward during catchup because catchup takes at least 10 min. During these 10 mins, it is very likely that new blocks are received and it would trigger new catchup jobs.

I think my implementation is just an approximation of initial catchup time. For this reason, I would defend my choice of only computing the time of first block being included into transition frontier as a good enough approximation.